### PR TITLE
nixos/make-options-doc: link manpages

### PR DESCRIPTION
**Pull request by ncfavier**
See discussion in https://github.com/NixOS/nixpkgs/pull/207268#issuecomment-1362940973, see commits.

In the future we could consider generating the mapping dynamically as a `.nix` file, e.g. extracting the full list of systemd manpages from `pkgs.systemd.man`.

If this is merged, nixos-search will temporarily have double links in nixos options because it extracts them from `options.json` (where the links have already been inserted) and [re-applies](https://github.com/NixOS/nixos-search/blob/24bdbc97b3e58d7d2056b3ead8bf8a5994f774fa/flake-info/src/data/pandoc.rs#L48) the old `link-unix-man-references.lua` filter in the Docbook → HTML conversion. Most browsers should handle this fine.

### CODE REVIEW

**Comment by Mic92** [comment]
Cool stuff!

**Review by jtojnar — APPROVED** [review]
Diff LGTM.

File: nixos/lib/make-options-doc/default.nix [hunk 1]
```diff
@@ -130,6 +130,8 @@ in rec {
         if baseOptionsJSON == null
         then builtins.toFile "base.json" "{}"
         else baseOptionsJSON;
+
+      NIXPKGS_PATH = pkgs.path;
```
Comment (ncfavier) [inline nixos/lib/make-options-doc/default.nix:134]: @roberth should this be `toString`ed or something to avoid copying?

Comment (roberth) [reply nixos/lib/make-options-doc/default.nix:134]: I don't think you need all of Nixpkgs. Could you pass the urls json instead?

Comment (ncfavier) [reply nixos/lib/make-options-doc/default.nix]: Yeah.

**Pull request merged**